pkg/chain/tbtc.go

```diff
@@ -93,14 +93,46 @@ type TBTCSystem interface {
 		handler func(depositAddress string),
 	) subscription.EventSubscription
 
+	// PastDepositCreatedEvents returns all created events for the given keep which
+	// occured after the provided start block. All implementations should return
+	// those events sorted by the block number in ascending order.
+	PastDepositCreatedEvents(
+		startBlock uint64,
+		keepAddress string,
+	) ([]*CreatedEvent, error)
+
 	// PastDepositRedemptionRequestedEvents returns all redemption requested
 	// events for the given deposit which occurred after the provided start block.
-	// All implementations should returns those events sorted by the
+	// All implementations should return those events sorted by the
 	// block number in the ascending order.
 	PastDepositRedemptionRequestedEvents(
 		startBlock uint64,
 		depositAddress string,
 	) ([]*DepositRedemptionRequestedEvent, error)
+
+	// DepositAddressForKeepAddress retrieves the deposit address for a particular keep address
+	DepositAddressForKeepAddress(
+		keepAddress string,
+	) (string, error)
```

Comment:
I would drop this and `PastDepositCreatedEvents` until you need them externally. That is, if `FundingInfo(depositAddress string) (*FundingInfo, error)` is the end goal here, that should be the only public method you add to the interface. You can keep the current implementations of `DepositAddressForKeepAddress` and `PastDepositCreatedEvents` in `ethereum`, `celo`, and `local` if it makes sense, but they can remain unexported.

Basically make sure the end goal you're aiming for is the only thing that introduces additional public API surface, as we're more or less committing to long-term support of that API surface.

Reply:
You'll see this play out in `local`, for example, where those extra two methods bring nothing to the table---and I don't think (but may be remembering wrong!) that the Bitcoin liquidation stuff requires more than just `FundingInfo`.

Reply:
Yeah - I started transitioning over to exposing `GetOwner`, and the first commit in that line is to just delete `PastDepositCreatedEvents` and `DepositAddressForKeepAddress`